src/storage/filesystem.py

```diff
@@ -185,23 +205,33 @@ def getMetadata(self, hashId):
                 ),
                 "r",
             ) as f:
-                objectType, streamNumber, expiresTime, tag, undef = string.split(f.read(), ",", 4)
-                return [int(objectType), int(streamNumber), int(expiresTime), unhexlify(tag)]
+                objectType, streamNumber, expiresTime, tag = string.split(
+                    f.read(), ",", 4)[:4]
+                return [
+                    int(objectType),
+                    int(streamNumber),
+                    int(expiresTime),
+                    unhexlify(tag)]
         except IOError:
             raise KeyError
 
     def by_type_and_tag(self, objectType, tag):
         """Get a list of objects filtered by object type and tag"""
         retval = []
-        for stream, streamDict in self._inventory:
+        for streamDict in self._inventory[1]:
```

Comment:
This wont work

Reply:
I am not tested this, but I thought this would work. btw thank you, i will change the soon

Reply:
So test. You iter through element 1 of inventory instead of inventory itself. It seems a mistake.